Signal/src/UIColor+OWS.m

```diff
@@ -54,6 +55,19 @@ + (UIColor *) ows_lightBackgroundColor {
     return [UIColor colorWithRed:242.f/255.f green:242.f/255.f blue:242.f/255.f alpha:1.f];
 }
 
++ (UIColor*) backgroundColorForContact:(NSString*)contactIdentifier {
+    NSArray *colors = @[[UIColor colorWithRed:213.f/255.f green:190.f/255.f blue:112.f/255.f alpha:1.f],
+                        [UIColor colorWithRed:18.f/255.f green:211.f/255.f blue:102.f/255.f alpha:1.f],
+                        [UIColor colorWithRed:32.f/255.f green:144.f/255.f blue:234.f/255.f alpha:1.f],
+                        [UIColor colorWithRed:180.f/255.f green:138.f/255.f blue:255.f/255.f alpha:1.f],
+                        [UIColor colorWithRed:255.f/255.f green:145.f/255.f blue:81.f/255.f alpha:1.f]];
+    NSData *contactData = [contactIdentifier dataUsingEncoding:NSUTF8StringEncoding];
+    NSData *hashData = [Cryptography  computeSHA256:contactData  truncatedToBytes:1];
```

Comment:
@abolishme will get 20 colors today, and the blue border will be removed
with white initials. he seems to already like it, so it sounds like we will
be, with the expanded set. as soon as he gets me a color list, I'll put it
in but probably read in the color set statically in a more general way

On Mon, Feb 9, 2015 at 11:04 AM, Frederic Jacobs <notifications@github.com>
wrote:

> In Signal/src/UIColor+OWS.m
> <https://github.com/WhisperSystems/Signal-iOS/pull/581#discussion_r24354831>
> :
>
> > @@ -54,6 +55,19 @@ + (UIColor *) ows_lightBackgroundColor {
> >      return [UIColor colorWithRed:242.f/255.f green:242.f/255.f blue:242.f/255.f alpha:1.f];
> >  }
> >
> > ++ (UIColor*) backgroundColorForContact:(NSString*)contactIdentifier {
> > +    NSArray *colors = @[[UIColor colorWithRed:213.f/255.f green:190.f/255.f blue:112.f/255.f alpha:1.f],
> > +                        [UIColor colorWithRed:18.f/255.f green:211.f/255.f blue:102.f/255.f alpha:1.f],
> > +                        [UIColor colorWithRed:32.f/255.f green:144.f/255.f blue:234.f/255.f alpha:1.f],
> > +                        [UIColor colorWithRed:180.f/255.f green:138.f/255.f blue:255.f/255.f alpha:1.f],
> > +                        [UIColor colorWithRed:255.f/255.f green:145.f/255.f blue:81.f/255.f alpha:1.f]];
> > +    NSData *contactData = [contactIdentifier dataUsingEncoding:NSUTF8StringEncoding];
> > +    NSData *hashData = [Cryptography  computeSHA256:contactData  truncatedToBytes:1];
>
> Gonna wait before merging. I don't think we're going to go down this road.
> Picking all these different colors is gonna look really bad but probably
> @abolishme <https://github.com/abolishme> can show us a mockup of how he
> wants it to be done instead.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/WhisperSystems/Signal-iOS/pull/581/files#r24354831>.
>



-- 
Dr. Corbett Moran
Physics @ http://www.tapir.caltech.edu <http://www.ICS.uzh.ch>
Dev @ http://circleof6app.com
Dev @ https://whispersystems.org
www.christinecorbettmoran.com